execution_engine/src/runtime_context/mod.rs

```diff
@@ -245,20 +284,21 @@ where
     /// also persistable map (one that is found in the
     /// TrackingCopy/GlobalState).
     pub fn remove_key(&mut self, name: &str) -> Result<(), Error> {
-        match self.base_key() {
+        match self.get_entity_address() {
             account_hash @ Key::Account(_) => {
-                let account: Account = {
-                    let mut account: Account = self.read_gs_typed(&account_hash)?;
-                    account.named_keys_mut().remove(name);
-                    account
+                let (contract, contract_key): (AddressableEntity, Key) = {
+                    let contract_key = self
+                        .read_gs_typed::<CLValue>(&account_hash)?
+                        .into_t::<Key>()?;
+
+                    let contract: AddressableEntity = self.read_gs_typed(&contract_key)?;
+                    (contract, contract_key)
```

Comment:
since addressableentity is not content addressable and is mutable, there's nothing preventing us from just sticking the addressableentityhash onto the addressableentity struct / making it self describing. this would simplify a lot of places where we juggle the hash or key + the record, as well as off chain consumers who also have to do it.